### PR TITLE
In conservative mode RedirectedThreadFrame should report registers to GC.

### PR DESCRIPTION
**Pull request by VSadov**
When a thread is redirected, its registers are stored in a context structure that is not on the stack.
The registers may contain managed roots and not reporting them to GC can cause crashes when running in conservative mode.

### CODE REVIEW

**Comment by mangod9** [comment]
was this found via a stress test or a CR?

**Comment by VSadov** [comment]
If you run `GC\Stress\Framework\ReliabilityFramework` test with `set COMPlus_gcConservative=1`, it will crash in about a minute due to this.

We do not generally test conservative mode. That is mostly for platform bring-up purposes. Or when investigating bugs, it can be enabled to rule out issues with gc root tracking (since it does not use any of that). 

It is supposed to work though. I guess - who uses it, fixes it.

File: src/coreclr/gc/gcconfig.h [hunk 1]
```diff
@@ -65,7 +65,7 @@ class GCConfigStringHolder
 #define GC_CONFIGURATION_KEYS \
     BOOL_CONFIG  (ServerGC,               "gcServer",               NULL,                             false,             "Whether we should be using Server GC")                                                   \
     BOOL_CONFIG  (ConcurrentGC,           "gcConcurrent",           NULL,                             true,              "Whether we should be using Concurrent GC")                                               \
-    BOOL_CONFIG  (ConservativeGC,         "gcConservative",         NULL,                             false,             "Enables/Disables conservative GC")                                                       \
+    BOOL_CONFIG  (ConservativeGC,         "gcConservative",         NULL,                             true,              "Enables/Disables conservative GC")                                                       \
```
Comment (VSadov) [inline src/coreclr/gc/gcconfig.h:68]: I temporarily enabled conservative GC by default - for a CI run. That is not a part of the fix, just for test coverage. I will undo config changes.

**Comment by VSadov** [comment]
The CI run with conservative mode enabled by default had some failures. 
However every failure I looked at was in a test that relies on precise GC (i.e. postmortem finalization, collectible assemblies, COM wrappers, etc). Such failures are bydesign in conservative mode. 

There are no random crashes anymore though, so I think the issue is fixed.

**Comment by VSadov** [comment]
@jkotas @davidwrighton - a fairly small change. Could you take a look?

The failures in tests are expected. They basically show that conservative mode works as expected. 
The failures will be gone once I undo test-only config changes.

**Review by jkotas — APPROVED** [review]

**Comment by VSadov** [comment]
Thanks!!

**Pull request merged**